File: java/src/televentas/PagoTarjetaCredito.java
package televentas;

import java.time.LocalDate;

public class PagoTarjetaCredito extends Pago {

    private final String numeroEnmascarado;
    private final String titular;
    private final String franquicia;
    private final String fechaVencimiento;

    public PagoTarjetaCredito(
            double monto,
            LocalDate fechaPago,
            String numeroEnmascarado,
            String titular,
            String franquicia,
            String fechaVencimiento
    ) {
        super(monto, fechaPago);
        this.numeroEnmascarado = numeroEnmascarado;
        this.titular = titular;
        this.franquicia = franquicia;
        this.fechaVencimiento = fechaVencimiento;
    }

    @Override
    public boolean validar() {
        return monto > 0
                && numeroEnmascarado != null && numeroEnmascarado.length() >= 4
                && titular != null && !titular.trim().isEmpty()
                && franquicia != null && !franquicia.trim().isEmpty()
                && fechaVencimiento != null && !fechaVencimiento.trim().isEmpty();
    }

    @Override
    public boolean procesar() {
        if (!validar()) {
            estado = "rechazado";
            return false;
        }
        estado = "aprobado";
        return true;
    }
}

File: javascript/src/televentas/pagoTarjetaCredito.js
import { Pago } from "./pago.js";

export class PagoTarjetaCredito extends Pago {
  constructor(monto, fechaPago, numeroEnmascarado, titular, franquicia, fechaVencimiento) {
    super(monto, fechaPago, "pendiente");
    this.numeroEnmascarado = numeroEnmascarado;
    this.titular = titular;
    this.franquicia = franquicia;
    this.fechaVencimiento = fechaVencimiento;
  }

  validar() {
    return (
      this.monto > 0 &&
      this.numeroEnmascarado != null &&
      this.numeroEnmascarado.length >= 4 &&
      this.titular != null &&
      this.titular.trim() !== "" &&
      this.franquicia != null &&
      this.franquicia.trim() !== "" &&
      this.fechaVencimiento != null &&
      this.fechaVencimiento.trim() !== ""
    );
  }

  procesar() {
    if (!this.validar()) {
      this.estado = "rechazado";
      return false;
    }
    this.estado = "aprobado";
    return true;
  }
}
